server/resources/resourcesController.js

```diff
@@ -63,6 +64,27 @@ router
 
 router
 	.route("/:id")
+	.get(
+		asyncHandler(async (req, res) => {
+			const {
+				rows: [resource],
+			} = await db.query(
+				singleLine`
+			SELECT r.*, t.name AS topic_name, us.name AS source_name, up.name AS publisher_name
+			FROM resources AS r
+			INNER JOIN users AS us ON r.source = us.id
+			LEFT JOIN users AS up ON r.publisher = up.id
+			LEFT JOIN topics AS t ON r.topic = t.id
+			WHERE r.id = $1;
+		`,
+				[req.params.id]
+			);
+			if (!resource) {
+				return res.sendStatus(404);
+			}
+			res.json(resource);
+		})
+	)
```

Comment:
Well written, thank you clear coding. Could you consider adding a try-catch block for error handling as well?